src/Util.cc

```diff
@@ -747,15 +750,29 @@ std::optional<math::Vector3d> getGridFieldCoordinates(
     }
   }
   auto position = origin->Data().PositionTransform(
-      _worldPosition, math::SphericalCoordinates::LOCAL2,
+      math::CoordinateVector3::Metric(_worldPosition),
+      math::SphericalCoordinates::LOCAL,
       _gridField->reference);
+  if (!position.has_value())
+    return std::nullopt;
+
+  if (position->IsMetric())
+    return position->AsMetricVector();
+
+  math::Vector3d out;
   if (_gridField->reference == math::SphericalCoordinates::SPHERICAL &&
     _gridField->units == components::EnvironmentalData::ReferenceUnits::DEGREES)
   {
-    position.X(GZ_RTOD(position.X()));
-    position.Y(GZ_RTOD(position.Y()));
+    out.X(position->Lat()->Degree());
```

Comment:
Hmm, it seems GridField could also be converted to CoordinateVector3 to explicitly express the units. However, I think it could be left for a future release of Gazebo...

Reply:
Yeah GridField should use CoordinateVector3 but we don't have time to update it right now.

Reply:
added a todo note for this. [9f1c0db](https://github.com/gazebosim/gz-sim/pull/2535/commits/9f1c0dbbcc50e89ae9a605811e9cbb18cc460f90)